cardano-api/src/Cardano/Api/Governance/Poll.hs

```diff
@@ -0,0 +1,391 @@
+-- Governance Poll Hash
+--
+
+newtype instance Hash GovernancePoll =
+    GovernancePollHash { unGovernancePollHash :: Hash.Hash (HASH StandardCrypto) GovernancePoll }
+  deriving stock (Eq, Ord)
+  deriving (Show, IsString) via UsingRawBytesHex (Hash GovernancePoll)
+
+instance SerialiseAsRawBytes (Hash GovernancePoll) where
+    serialiseToRawBytes =
+      hashToBytes . unGovernancePollHash
+
+    deserialiseFromRawBytes (AsHash AsGovernancePoll) bs =
+      maybeToRight (SerialiseAsRawBytesError "Unable to deserialise Hash(GovernancePoll)") $
+        GovernancePollHash <$> hashFromBytes bs
+
+hashGovernancePoll :: GovernancePoll -> Hash GovernancePoll
+hashGovernancePoll =
+  GovernancePollHash . hashWith @(HASH StandardCrypto) serialiseToCBOR
+
+
+-- ----------------------------------------------------------------------------
+-- Governance Poll Answer
+--
+
+data GovernancePollAnswer = GovernancePollAnswer
+    { govAnsPoll :: Hash GovernancePoll
+    , govAnsChoice :: Word
+    }
+  deriving Show
+
+instance HasTypeProxy GovernancePollAnswer where
+    data AsType GovernancePollAnswer = AsGovernancePollAnswer
+    proxyToAsType _ = AsGovernancePollAnswer
+
+instance SignableRepresentation GovernancePollAnswer where
+    getSignableRepresentation =
+      hashToBytes . hashWith @(HASH StandardCrypto) (serialiseToCBOR . asTxMetadata)
+
+instance AsTxMetadata GovernancePollAnswer where
+    asTxMetadata GovernancePollAnswer{govAnsPoll, govAnsChoice} =
+      makeTransactionMetadata $ Map.fromList
+        [ ( metadataLabel
+          , TxMetaMap $
+           [ ( TxMetaNumber 2, TxMetaBytes (serialiseToRawBytes govAnsPoll) )
+           , ( TxMetaNumber 3, TxMetaNumber (toInteger govAnsChoice) )
+           ]
+          )
+        ]
+
+instance SerialiseAsCBOR GovernancePollAnswer where
+    serialiseToCBOR =
+      serialiseToCBOR . asTxMetadata
+
+    deserialiseFromCBOR AsGovernancePollAnswer bs = do
+      metadata <- deserialiseFromCBOR AsTxMetadata bs
+      withNestedMap lbl metadataLabel metadata $ \values ->
+        GovernancePollAnswer
+          <$> case lookup (TxMetaNumber 2) values of
+                Nothing -> Left $ missingField (field "2")
+                Just x  -> expectHash (field "2") x
+          <*> case lookup (TxMetaNumber 3) values of
+                Just (TxMetaNumber n) -> expectWord (field "3") n
+                Just _  -> Left $ malformedField (field "3") "Number (uint)"
+                Nothing -> Left $ missingField (field "3")
+     where
+       lbl = "GovernancePollAnswer"
+       field i = lbl <> "." <> i
+
+       expectHash fieldLbl value =
+         case value of
+           TxMetaBytes bytes ->
+             left
+               (DecoderErrorCustom fieldLbl . Text.pack . unSerialiseAsRawBytesError)
+               (deserialiseFromRawBytes (AsHash AsGovernancePoll) bytes)
+           _ ->
+             Left (malformedField fieldLbl "Bytes")
```

Comment:
Should we say more than `"Bytes"`?

Reply:
https://github.com/input-output-hk/cardano-node/compare/c018de7877eb6c513ddc029224ba2a189c4f6b2c..7adb11180e227089a5aa44845a06737efacf3401